EQP updates (#5918)
magento/devdocs: marketplace/eqp/v1/packages.md

title: Packages
---

Use this resource to initiate and manage all aspects of submitting a package to the
[Magento Extension Quality Program (EQP)](http://docs.magento.com/marketplace/user_guide/extensions/extension-quality-program.html).
You can provide all metadata associated with a package, both the technical and the marketing information, in a single step, or in several steps, using incremental updates.

> *  **Technical information** - References to code artifacts, such as Magento 1 TGZ files or Composer-compliant Magento 2 ZIP files, version compatibility, and release notes.

Reviewer: The original em dash was correct (no spaces on either side).

> *  **Technical information** - References to code artifacts, such as Magento 1 TGZ files or Composer-compliant Magento 2 ZIP files, version compatibility, and release notes.

Reviewer: ```suggestion
*  **Technical information**—References to code artifacts, such as Magento 1 TGZ files or Composer-compliant Magento 2 ZIP files, version compatibility, and release notes.
```

> *  **Marketing information** - Includes package descriptions, image assets for logos and galleries, pricing information, support and installation services offered, and various guides (user, installation, and reference) in PDF.

Reviewer: Same here

> *  **Marketing information** - Includes package descriptions, image assets for logos and galleries, pricing information, support and installation services offered, and various guides (user, installation, and reference) in PDF.

Reviewer: ```suggestion
*  **Marketing information**—Includes package descriptions, image assets for logos and galleries, pricing information, support and installation services offered, and various guides (user, installation, and reference) in PDF.
```

You can submit a package in either of the following ways:

*  A single POST request with all required fields set. You must explicitly indicate that you are submitting for technical and marketing review using the `action` parameter.
*  A series of requests, typically in the following order:
   1. A single POST request with the minimum required fields set and `action` set to `draft` in either `technical`, `marketing`, or both. This request accepts the new package and saves it on the Developer Portal for further updates. It returns a unique `submission_id` for subsequent PUT operations.
   1. One or more PUT requests in which you configure the package parameters. In these requests, set `action` to `draft` in `technical`, `marketing`, or both.
   1. A final PUT request indicating submission for `technical`, `marketing`, or both.

You can update one or more parameters in `draft` mode.

> In this mode, the API checks only for basic type-validation issues.

Reviewer: ```suggestion
In this mode, the API checks for basic type-validation issues only.
```

```

*  The API returns a HTTP 200 batch response listing items in the same order as they were provided in the request.
*  Each item contains a `code` and `message` indicating success or failure. Any non-200 code indicates an error.
   Refer to the message for more details.
*  A unique `submission_id` is returned for each successful item, which must be used for any GET or PUT methods.
*  Optionally, if a user-defined `item_id` was supplied during the POST,

>    the response will echo back the same `item_id` for each item in the batch.

Reviewer: ```suggestion
   the response will return the same `item_id` for each item in the batch.
```

   The resource can be retrieved via GET using the `item_id`.
*  Any non-200 HTTP response code indicates an error for the entire batch request.

### Update a package

The PUT method can be used to update packages in the following states:

*  The package is in draft mode for the technical or marketing review; or both.
*  The package has been rejected in either the technical or marketing review; or both.
   You must fix these issues and re-submit the package.
*  The package has been released to the Magento Marketplace.
*  The package was removed from the store by the developer and needs to be re-published.
*  The package can be recalled while in the EQP pipeline.

> *  After a package has been released to the Magento Marketplace, you can update marketing information only.

Reviewer: ```suggestion
*  After a package has been released on the Magento Marketplace, you can update marketing information only.
```

>    Changing marketing information causes the package to be placed in marketing review.

Reviewer: ```suggestion
   Changes to marketing information require marketing review.
```

>    The package continues to be live on the marketplace, and after the marketing approval,

Reviewer: ```suggestion
   The package continues to be live on the marketplace, and after approval,
```

curl -X PUT \
     -H 'Authorization: Bearer baGXoStRuR9VCDFQGZNzgNqbqu5WUwlr.cAxZJ9m22Le7' \
     -H 'Content-Type: application/json' \
     -d '{ "action" : { "overall" : "publish"} }' \
     https://developer-api.magento.com/rest/v1/products/packages/f4eacd72be
```

The HTTP response code will indicate success or failure.

#### Required parameters

> If the **action** parameter gives a `submit` value for technical, marketing or both,

Reviewer: ```suggestion
If the **action** parameter contains a `submit` value for technical, marketing or both,
```
